### ISSUE TITLE
Many tests may be flaky after #10198

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt#10906]
https://github.com/w3c/web-platform-tests/pull/10198 was blocked for a long time because of being flaky in Chrome Dev, which was never resolved. It was finally force merged because the changes had been reverted in Chromium by automatic import, and the best way to recover was to merge the PR.

The last Travis job showing the flakiness:
https://travis-ci.org/w3c/web-platform-tests/jobs/376549386

**Comment by foolip** [comment]
OK, "good", https://github.com/w3c/web-platform-tests/pull/10907 is still flaky when touching the tests, so this is possible to investigate.

**Comment by Honry** [comment]
@foolip, the flakiness disappeared in my PR https://github.com/w3c/web-platform-tests/pull/10962, I didn't do much change, but it's magic.  :)

But most tests still failed in my local, so I did an investigation and have some findings.

1.  See following code snippet from [generic-sensor-tests.js](https://github.com/w3c/web-platform-tests/blob/master/generic-sensor/generic-sensor-tests.js#L47
), the code `await sensorTest.reset()` never been invoked once tests failed at `await func(t)`, which would lead to failure of mock sensor initialization in following tests.  

```
function sensor_test(func, name, properties) {
  promise_test(async (t) => {
    if (typeof GenericSensorTest === 'undefined') {
      await loadChromiumResources;
    }
    assert_true(typeof GenericSensorTest !== 'undefined');
    let sensorTest = new GenericSensorTest();
    await sensorTest.initialize();
    try {
      await func(t);
    } finally {
      await sensorTest.reset();
    };
  }, name, properties);
}
```
This can be fixed by using `add_cleanup` function support for `promise_test`, but which is still under review at https://github.com/w3c/web-platform-tests/pull/8748

2. Based on fixing in item [1], I found that the mock sensor reading is always 0, @alexshalamov, any idea?

**Comment by rakuco** [comment]
> But most tests still failed in my local, so I did an investigation and have some findings.
> 
> 1. See following code snippet from generic-sensor-tests.js, the code await sensorTest.reset() never been invoked once tests failed at await func(t), which would lead to failure of mock sensor initialization in following tests.

I believe this one's been fixed by #11625 and we're no longer blocked on `add_cleanup` + promises.

> 2. Based on fixing in item [1], I found that the mock sensor reading is always 0, @alexshalamov, any idea?

This one is being tracked in https://bugs.chromium.org/p/chromium/issues/detail?id=853132, but the tests relying on `Sensor.reading` should fail consistently instead of being flaky. 

Can we close this bug?

**Comment by Honry** [comment]
Cheers, let's close it.
